disallowKeywordsOnNewLine: Allow comments before keywords

oredi opened the pull request:
Allow a special case for the keywords 'else' in 'if else' and 'while' in 'do while' to go on a new line if it is preceded by a comment token

Fixes #1421

mrjoelkemp commented on test/specs/rules/disallow-keywords-on-new-line.js:
```diff
@@ -130,6 +130,37 @@ describe('rules/disallow-keywords-on-new-line', function() {
 
+    describe('legal comments before key word (#1421)', function() {
+        it('should not report legal keyword placement for a "if else"',
+            function() {
```
this was done for minimumLineLength? Would like this on the previous line

mrjoelkemp commented:
Nice work here @oredi! 

Can we have a test for a single line comment? Still not seeing why the regex was used – a test would help clarify the comment.

oredi commented:
@mrjoelkemp I added the tests for Single comment and removed the regex. I was having issues initially with single line comments as I did not place a line break after the comment. So that commented out the test code after the single line comment. After looking into the other rules for comments, I think I got it right now.

mrjoelkemp commented:
lgtm. nice work!

markelog commented:
Thank you!
